Small fix for locked boost conversion priority

JConquistador opened the pull request.

JConquistador committed:
Fix locked boost converstion in InventoryManager.cs

Didn't test, just doing some tweaking

JConquistador committed:
Small typo fixes

JConquistador committed:
1) Adding a debug log
2) disabling a titan from either the gold or titan list if they fail to autokill within 60 seconds of becoming active
3) Adding support for walderp manual combat

JConquistador committed:
Fixing bug with BR-X (X was compared against ritual endtime including the rebirth time)
Also added a check for BR to skip if the next breakpoint would arrive before the ritual completion

JConquistador committed:
Augments and Blood Magic require gold to run. If we dont have enough gold, any energy/magic assigned is wasted

Update BESTAUG, AUG-X, BR, and RIT-X to return false from Allocate call if no resources are assigned. Also added gold check to AUG-X.

Implemented loop in AllocateEnergy and AllocateMagic. If any Allocate call fails, remove it from the list of priorites and try again.

JConquistador committed:
Merge pull request #1 from JConquistador/JConqTinkering

J conq tinkering

JConquistador committed:
1) Fixing walderp combat bug
2) Heals more frequently and in line with fast combat heals, also continue buffing and healing the entire time instead of stopping in the final 20% when in titan zones
3) Increasing titan disabling check to 5 minutes from 1, dont disable if currently doing combat in a titan zone
4) Bound Reset Boost Average button to the Reset method

JConquistador committed:
changing item stat values to floats from decimals to match the underlying equipment types, a little debug logging, never allow a calculated boost value to go below 0 when determining needed boosts

JConquistador committed:
Had to update .csproj file for new BloodMagic class reference, this also updates the Unity/.net framework/steam path references! Windows forms references also updated to 4.0 on the SettingsForm.resx

1) New feature to cast a specific blood spell on rebirth even if the threshold isnt met. Priority is GuffB > Guff A > Iron Pill > Number
2) Fixed issue with Diggers where if there isnt enough GPS to enable the configured diggers they'd stay stuck as inactive until the next breakpoint

JConquistador committed:
Merge pull request #2 from JConquistador/JConqTinkering

J conq tinkering

q952087842 commented:
Hi bro, could you please fix the boss's combat mechanism? It was terrible in the challenge，will waste a lot of time.The code located in NGUInjector/Main.cs，if (bc.character.attack > (bc.character.bossDefense * 1.4) && bc.character.defense > bc.character.bossAttack * 1.4).In the initial project, I saw someone raise this issue and modify it to if (bc.character.defense/20-bc.character.bossAttack+bc.character.defense > bc.character.bossDefense/20-bc.character.attack+bc.character.bossDefense)。I have tried to fix this problem myself, but I cannot find a way to modify it and successfully run it.

JConquistador commented:
> Hi bro, could you please fix the boss's combat mechanism? It was terrible in the challenge，will waste a lot of time.The code located in NGUInjector/Main.cs，if (bc.character.attack > (bc.character.bossDefense * 1.4) && bc.character.defense > bc.character.bossAttack * 1.4).In the initial project, I saw someone raise this issue and modify it to if (bc.character.defense/20-bc.character.bossAttack+bc.character.defense > bc.character.bossDefense/20-bc.character.attack+bc.character.bossDefense)。I have tried to fix this problem myself, but I cannot find a way to modify it and successfully run it.

I'll check it out and try and get this into my branch.

JConquistador committed:
Fixing a bug for needed boosts

JConquistador committed:
Changing boss attack logic to use the actual number of attacks to kill. If the character will kill in fewer attacks than the boss, enable the fight. Also some minor logging changes.

JConquistador committed:
Merge pull request #3 from JConquistador/JConqTinkering

J conq tinkering

JConquistador commented:
> Hi bro, could you please fix the boss's combat mechanism?

Ok, I did an update to fight the boss at the earliest possible moment. Instead of using a formula I use the game's boss damage logic to determine whether the character or the boss has the fewest number of attacks to kill. Release with this update is posted on my fork:
https://github.com/JConquistador/NGUInjector/releases

q952087842 commented:
oh,I tried, it was very well.

JConquistador committed:
cast blood spells for no rebirth profiles

JConquistador committed:
refactoring, cleaning up and fixing the various item pickers

JConquistador committed:
GUI support for quest loadout items

JConquistador committed:
Add missing label for ygg tab

JConquistador committed:
set item labels upon launching

JConquistador committed:
Minor tweak to UI positioning, update readme

JConquistador committed:
ui fix for quest items

JConquistador committed:
Merge pull request #4 from JConquistador/JConqTinkering

J conq tinkering

JConquistador committed:
update version

JConquistador committed:
small UI positioning on quest tab, version update

JConquistador committed:
Merge pull request #5 from JConquistador/JConqTinkering

J conq tinkering

JConquistador committed:
- Fixing bat file to allow for running as admin
- Move settings folder under appdata instead of the desktop
- Add some buttons to the UI to open the settings/profile folders

JConquistador committed:
Merge pull request #6 from JConquistador/JConqTinkering

- Fixing bat file to allow for running as admin

JConquistador committed:
making note bold

JConquistador committed:
blood thresholds should be <=

JConquistador committed:
fix for beast mode toggle in idle mode

JConquistador committed:
additional refactoring to have beast mode handled by the same method, stay in safe zone while waiting for beast mode if in idle combat

JConquistador committed:
real fix for autospell threshold comparisons (float vs int)

JConquistador committed:
remove log debug

JConquistador committed:
Merge pull request #7 from JConquistador/JConqTinkering

J conq tinkering

JConquistador committed:
update to rc4

JConquistador committed:
Merge pull request #8 from JConquistador/JConqTinkering

update to rc4

JConquistador committed:
minor fix to beast mode toggle and use the idle combat loop if no attacks have been unlocked yet in basic training

JConquistador committed:
fight tracking for idle (fixes bug with gold snipe before unlocking regular attack)

JConquistador committed:
Merge pull request #9 from JConquistador/JConqTinkering

J conq tinkering

JConquistador committed:
use enemyAI for walderp logic instead of log parsing, intelligently use cooldown times for number of titan attacks to hold skill use, beta untested version of Godmother combat

JConquistador committed:
Merge pull request #10 from JConquistador/JConqTinkering

use enemyAI for walderp logic instead of log parsing, intelligently u…

JConquistador committed:
block used to quickly will expire while godmother is exploding, wait a little bit

JConquistador committed:
tweak to godmother wait time

JConquistador committed:
squeeze in one more attack in godmother combat while delaying for block

JConquistador committed:
fix bug where not holding block while waiting for right timing on explosion attack

JConquistador committed:
fix to wish list in UI to allow for the first wish, if the current wish is hidden and has no resources assigne to it, select a wish from the wish BP list

JConquistador committed:
Ignore bossonly flag for titans, allow combat in multiple titan zones at once with new "Titans" flag in Adventure tab

JConquistador committed:
Update to readme and disable titan combat after 5 minutes regardless of if the current combat zone is set to the titan's zone or not

JConquistador committed:
Merge pull request #11 from JConquistador/JConqTinkering

J conq tinkering

JConquistador committed:
Refresh titan snapshots on each combat loop, not each main loop

JConquistador committed:
New smart beast mode, titan specific combat settings, UI provides more information about which combat settings are used for different combat modes

JConquistador committed:
Merge pull request #12 from JConquistador/JConqTinkering

J conq tinkering

JConquistador committed:
move zones to zone helpers, prep for 3.7.1 release

JConquistador committed:
selecting an unhidden wish does not play nicely with pagination, remove the feature

JConquistador committed:
log combat kill a bit earlier, bug fix for smartbeastmode precasting

JConquistador committed:
refactoring some combat, break out of smart beast mode if we cant cast pierce or ultimate before timer is up

JConquistador committed:
Merge pull request #13 from JConquistador/JConqTinkering

J conq tinkering

JConquistador committed:
A real fix for the wish menu, new wish will be selected the currently selected wish is filtered, page will not change when the menu is refreshed in the main loop by the allocation profiles

JConquistador committed:
better handling of all NumericUpDown controls

JConquistador committed:
handle wandoos cap can be negative on nongu challenge

JConquistador committed:
remove log lines

JConquistador committed:
Exile combat support and major rework to defensive cooldown logic

JConquistador committed:
update version string

JConquistador committed:
Some notes about future TODOs, minor fix for AdvStat purchase, attempting to guess Exile v2/3/4 combat quirks

JConquistador committed:
Adding muffin rebirth, improving consumable use logic to not overuse consumables if restarting the game or reloading a profile, cleaning up move69, removed some unused settings, tweak to combat to take movetimer into consideration (0.8f if red liquid maxxed, 1.0f otherwise)

JConquistador committed:
work generic enemy ai special attacks into the same godmother/exile block logic

JConquistador committed:
update version tag

JConquistador committed:
Merge pull request #14 from JConquistador/JConqTinkering

J conq tinkering

JConquistador committed:
Rework combat (again), nerd combat support, adding wishblacklist and card sorting to UI

JConquistador committed:
clean up of warning placement, wish blacklist bug fix

JConquistador committed:
final testing and fixes for muffin rebirth, locate walderp button, update readme with new screenshots

JConquistador committed:
Merge pull request #15 from JConquistador/JConqTinkering

J conq tinkering

JConquistador committed:
zone no longer necessary in combat snapshot

JConquistador committed:
some error handling

JConquistador committed:
change muffinrebirth slightly to allow for configured rebirth time, also add a "timebalancedmuffin" rebirth which adds the same amount of time to the longer cycle as it takes from the shorter cycle

JConquistador committed:
Merge pull request #16 from JConquistador/JConqTinkering

J conq tinkering

JConquistador committed:
beta testing of exilev2, tightening up defensive cooldown logic

JConquistador committed:
Multiple mayo generators now function better. Prioritizes generating mayo for any cards missing mayo in order of the cards to be cast. If any mayo generators are open after all pending cards are accounted for mayo will generate according to the mayo with the lowest amount and then lowest progress.

JConquistador committed:
Revert "Multiple mayo generators now function better. Prioritizes generating mayo for any cards missing mayo in order of the cards to be cast. If any mayo generators are open after all pending cards are accounted for mayo will generate according to the mayo with the lowest amount and then lowest progress."

This reverts commit 4a3af6a120c227997efc6616c58ae5c7178e2534.

JConquistador committed:
fix bug where titan gear was stuck as locked after gold kill

JConquistador committed:
incorporate TC2 check into MuffinRB

JConquistador committed:
Final tweaks to combat for TitanV9, bugfix for Muffin based rebirthing

JConquistador committed:
Merge pull request #17 from JConquistador/JConqTinkering

J conq tinkering

JConquistador committed:
Dont paralyze enemies in rapid mode (rapid type AI and Godmother combat)

JConquistador committed:
update version num

JConquistador committed:
Merge pull request #18 from JConquistador/JConqTinkering

J conq tinkering

JConquistador committed:
- Implementation of Cooking tab
- New flag on Cards tab for users to decide if protected cards should be trashed
- Check the inventory (not accessories) when checking if we should trash an END card
- Fixed bug involving trashing protected END cards
- Cards are trashed in reverse order instead of trashing from the first to the last
- Prevent idling in ITOPOD in quest gear when sniping a titan zone
- Clarification on wording on titans tab

JConquistador committed:
Merge pull request #19 from JConquistador/JConqTinkering

- Implementation of Cooking tab

JConquistador committed:
fix for flickering quest text

JConquistador committed:
Merge pull request #20 from JConquistador/JConqTinkering

fix for flickering quest text

JConquistador committed:
chonkers are always protected, make sure they can be trashed if the flag is set

JConquistador committed:
cleaning up card trash logic

JConquistador committed:
trashing chonkers takes priority over not trashing protected

JConquistador committed:
Merge pull request #21 from JConquistador/JConqTinkering

J conq tinkering

JConquistador committed:
Fix for charger and rapid ai enemy type loops, fix for wish menu update causing crash when no wishes visible

JConquistador committed:
Merge pull request #22 from JConquistador/JConqTinkering

Combat and wish bug fixes,

JConquistador committed:
Smarter use of Charge skill, always try to save for ultimate attack

JConquistador committed:
Merge pull request #23 from JConquistador/JConqTinkering

Smarter use of Charge skill, always try to save for ultimate attack

JConquistador committed:
adding a difficulty changer to the titan tab

rus9384 committed:
Update QuestManager.cs

- Allowed to start idle minors while in adventure.
- Added a UI update after starting quests.

rus9384 committed:
Update WandoosBP.cs

Fixed wandoos allocation

rus9384 committed:
Update LoadoutManager.cs

Fixed resource amounts after swapping gear

JConquistador committed:
Merge pull request #24 from rus9384/master

Small bug fixes

JConquistador committed:
Merge branch 'JConqTinkering' of https://github.com/JConquistador/NGUInjector into JConqTinkering

JConquistador committed:
Adding softcap option to cube boost priority, small fix on preselecting titan version upon startup

JConquistador committed:
fix for wandoos, adding all the profiles i saved
